worker/convert/convert.go

```diff
@@ -98,3 +85,66 @@ func getImageDimensions(path string) (int, int) {
 	}
 	return width, height
 }
+
+func isAnimateable(path string) bool {
+
+	file, err := os.ReadFile(path)
+	if err != nil {
+		fmt.Printf("can not check if video length is animatable: %v", err)
+	}
+	if len(file) > animatableVideoLen {
+		return false
+	}
+	return true
+}
+
+func convertImage(task ConvertTask) error {
+
+	// FIXME: converting to webp's 512x512 skews aspect ratio
+	// So Find a way to convert to 512x512 while maintaining perspective before cwebp convertion
+
+	// Convert Image to WebP
+	// Using https://developers.google.com/speed/webp/docs/cwebp
+	cmd := *exec.Command("cwebp", task.MediaPath, "-resize", "0", "600", "-o", task.ConvertedPath)
+	err := cmd.Run()
+
+	return err
+}
+
+func convertVideo(task ConvertTask) error {
+
+	var qValue int
+	switch {
+	case task.DataLen < 350000:
+		qValue = 20
+	case task.DataLen < 650000:
+		qValue = 15
+	default:
+		qValue = 12
+	}
+	fmt.Printf("Q value is %d\n", qValue)
+	commandString := fmt.Sprintf("ffmpeg -i %s  -filter:v fps=fps=20 -compression_level 0 -q:v %d -loop 0 -preset picture -an -vsync 0 -s 800:800  %s", task.MediaPath, qValue, task.ConvertedPath)
+	cmd := *exec.Command("bash", "-c", commandString)
+	var outb, errb bytes.Buffer
+	cmd.Stdout = &outb
+	cmd.Stderr = &errb
+
+	err := cmd.Run()
+
+	// validate converted video is the right size
+	if !(isAnimateable(task.ConvertedPath)) {
+		fmt.Println("Reconverting video..\n")
+		os.Remove(task.ConvertedPath)
```

Comment:
> No need to remove convertedPath.... running the command should overwrite it there

it won't override it, that's why the reconverted videos were not sent because FFmpeg will prompt you to override the existing file hence it was stuck on that prompt.
 I'll update the command though to automatically override the existing file